pageserver/src/tenant/ephemeral_file/page_caching.rs

```diff
@@ -208,21 +208,13 @@ impl PreWarmingWriter {
 }
 
 impl crate::virtual_file::owned_buffers_io::write::OwnedAsyncWriter for PreWarmingWriter {
-    async fn write_all<
-        B: tokio_epoll_uring::BoundedBuf<Buf = Buf>,
-        Buf: tokio_epoll_uring::IoBuf + Send,
-    >(
+    async fn write_all<Buf: tokio_epoll_uring::IoBuf + Send>(
         &mut self,
-        buf: B,
+        buf: Slice<Buf>,
         ctx: &RequestContext,
-    ) -> std::io::Result<(usize, B::Buf)> {
-        let buf = buf.slice(..);
-        let saved_bounds = buf.bounds(); // save for reconstructing the Slice from iobuf after the IO is done
-        let check_bounds_stuff_works = if cfg!(test) && cfg!(debug_assertions) {
-            Some(buf.to_vec())
-        } else {
-            None
-        };
+    ) -> std::io::Result<(usize, Slice<Buf>)> {
+        assert_eq!(buf.bytes_init(), buf.bytes_total());
```

Comment:
idea: Wondering if we could have a `FullSlice<Buf>` type instead of sprinkling asserts. It would also remove the risk of having missed a call site somewhere.

Reply:
done in https://github.com/neondatabase/neon/pull/8717/commits/21573e3892be7a228ec2f347e21fe423110f6dca